test/e2e/stackdriver_monitoring.go

```diff
@@ -17,27 +17,104 @@ limitations under the License.
 package e2e
 
 import (
-	"k8s.io/kubernetes/test/e2e/framework"
+	"fmt"
+	"time"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
 
 	. "github.com/onsi/ginkgo"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/kubernetes/test/e2e/framework"
 
 	gcm "google.golang.org/api/monitoring/v3"
 )
 
-var _ = &gcm.CollectdValue{}
-
 var _ = framework.KubeDescribe("Stackdriver metrics", func() {
-	_ = framework.NewDefaultFramework("monitoring")
-
 	BeforeEach(func() {
 		framework.SkipUnlessProviderIs("gke")
 	})
 
 	It("should publish metrics to Stackdriver [Feature:Stackdriver][Flaky]", func() {
 		framework.Logf("Start creating Stackdriver monitoring test")
-		testStackdriverMonitoring(f.ClientSet)
+		testStackdriverMonitoring()
 	})
 })
 
+var (
+	stackdriverMetrics = []string{
+		"uptime",
+		"cpu/reserved_cores",
+		"cpu/usage_time",
+		"cpu/utilization",
+		"memory/bytes_total",
+		"memory/bytes_used",
+		"memory/page_fault_count",
+		"disk/bytes_used",
+		"disk/bytes_total",
+	}
+)
+
 func testStackdriverMonitoring() {
+	projectId := framework.TestContext.CloudConfig.ProjectID
+	pollingFunction := checkForMetrics(projectId, time.Now())
+	framework.ExpectNoError(wait.Poll(time.Second*5, time.Minute*7, pollingFunction))
+}
+
+func checkForMetrics(projectId string, start time.Time) func() (bool, error) {
+	return func() (bool, error) {
+		for _, metric := range stackdriverMetrics {
+			// TODO: check only for metrics from this cluster
+			ts, err := fetchTimeSeries(projectId, metric, start)
+			if err != nil {
+				framework.Failf("Error fetching %v, %v", metric, err)
+			}
+			if len(ts) < 1 {
+				return false, nil
+			}
+		}
+		return true, nil
+	}
+}
+
+func createMetricFilter(metric string) string {
+	return fmt.Sprintf("metric.type=\"container.googleapis.com/container/%s\"", metric)
+}
+
+func fetchTimeSeries(projectId string, metric string, start time.Time) ([]*gcm.TimeSeries, error) {
+	ts := google.ComputeTokenSource("")
+
+	// Hack for running tests locally
+	// If this is your use case, create application default credentials:
+	//
+	// $ gcloud auth application-default login
+	//
+	// and uncomment following lines:
+	//
+	// ts, err := google.DefaultTokenSource(oauth2.NoContext)
+	// framework.Logf("Couldn't get application default credentials, %v", err)
+	// if err != nil {
+	// 	framework.Failf("Error accessing application default credentials, %v", err)
+	// }
+
+	client := oauth2.NewClient(oauth2.NoContext, ts)
+	gcmService, err := gcm.New(client)
+	if err != nil {
+		return nil, err
+	}
+	response, err := gcmService.Projects.TimeSeries.
+		List(fullProjectName(projectId)).
+		Filter(createMetricFilter(metric)).
+		IntervalStartTime(start.Format(time.RFC3339)).
+		IntervalEndTime(time.Now().Format(time.RFC3339)).
+		Do()
+	if err != nil {
```

Comment:
It should be ok to just use err:
```
framework.ExpectNoError(err)
```
https://github.com/kubernetes/kubernetes/blob/61b7b3fb660d09917af0b843c1b151279ea574ef/test/e2e/metrics_grabber_test.go#L38